api/actions/serializers.py

```diff
@@ -315,3 +323,49 @@ def create(self, validated_data):
             )
 
         return target.actions.last()
+
+
+class SchemaResponseActionSerializer(BaseActionSerializer):
+    class Meta:
+        type_ = 'schema-response-actions'
+
+    permissions = ser.ChoiceField(choices=permissions.API_CONTRIBUTOR_PERMISSIONS, required=False)
+    visible = ser.BooleanField(default=True, required=False)
+    trigger = ser.ChoiceField(choices=SchemaResponseTriggers.char_field_choices())
+
+    target = TargetRelationshipField(
+        target_class=SchemaResponse,
+        read_only=False,
+        required=True,
+        related_view='schema_responses:schema-response-action-list',
+        related_view_kwargs={'schema_response_id': '<target._id>'},
+        filter_key='target__guids___id',
+    )
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+        trigger = validated_data.get('trigger')
+        target = validated_data.pop('target')
+        comment = validated_data.pop('comment', '')
+        try:
+            if trigger == SchemaResponseTriggers.SUBMIT.db_name:
+                required_approvers = [user.id for user, node in target.parent.get_admin_contributors_recursive(unique_users=True)]
+                target.submit(user=user, comment=comment, required_approvers=required_approvers)
+            elif trigger == SchemaResponseTriggers.APPROVE.db_name:
+                target.approve(user=user, comment=comment)
+            elif trigger == SchemaResponseTriggers.ACCEPT.db_name:
+                target.accept(user=user, comment=comment)
+            elif trigger == SchemaResponseTriggers.ADMIN_REJECT.db_name:
+                target.reject(user=user, comment=comment)
+            elif trigger == SchemaResponseTriggers.MODERATOR_REJECT.db_name:
+                target.reject(user=user, comment=comment)
+            else:
+                raise JSONAPIAttributeException(attribute='trigger', detail='Invalid trigger.')
+        except PermissionsError as exc:
+            raise PermissionDenied(exc)
+        except ValueError as exc:
+            raise ValidationError(exc)
+        except MachineError as exc:
+            raise Conflict(exc)
+
+        return target.actions.last()
```

Comment:
In the RegistrationAction serializer we do a check to ensure that the created Action has the same trigger as what was passed here. That's to avoid issues like what we see in some of your tests below where an admin user can *technically* issue a MODERATOR_REJECT trigger.

Please replicate that functionality here.